asciidtype/asciidtype/src/casts.c

```diff
@@ -102,3 +287,59 @@ PyArrayMethod_Spec ASCIIToASCIICastSpec = {
         .dtypes = a2a_dtypes,
         .slots = a2a_slots,
 };
+
+static PyType_Slot u2a_slots[] = {
+        {NPY_METH_resolve_descriptors, &unicode_to_ascii_resolve_descriptors},
+        {_NPY_METH_get_loop, &unicode_to_ascii_get_loop},
+        {0, NULL}};
+
+static char *u2a_name = "cast_Unicode_to_ASCIIDType";
+
+static PyType_Slot a2u_slots[] = {
+        {NPY_METH_resolve_descriptors, &ascii_to_unicode_resolve_descriptors},
+        {_NPY_METH_get_loop, &ascii_to_unicode_get_loop},
+        {0, NULL}};
+
+static char *a2u_name = "cast_ASCIIDType_to_Unicode";
+
+PyArrayMethod_Spec **
+get_casts(void)
+{
+    PyArray_DTypeMeta **u2a_dtypes = malloc(2 * sizeof(PyArray_DTypeMeta *));
+    u2a_dtypes[0] = &PyArray_UnicodeDType;
+    u2a_dtypes[1] = NULL;
+
+    PyArrayMethod_Spec *UnicodeToASCIICastSpec =
+            malloc(sizeof(PyArrayMethod_Spec));
+
+    UnicodeToASCIICastSpec->name = u2a_name;
+    UnicodeToASCIICastSpec->nin = 1;
+    UnicodeToASCIICastSpec->nout = 1;
+    UnicodeToASCIICastSpec->flags = NPY_METH_SUPPORTS_UNALIGNED;
+    UnicodeToASCIICastSpec->casting = NPY_SAME_KIND_CASTING;
+    UnicodeToASCIICastSpec->dtypes = u2a_dtypes;
+    UnicodeToASCIICastSpec->slots = u2a_slots;
+
+    PyArray_DTypeMeta **a2u_dtypes = malloc(2 * sizeof(PyArray_DTypeMeta *));
+    a2u_dtypes[0] = NULL;
+    a2u_dtypes[1] = &PyArray_UnicodeDType;
+
+    PyArrayMethod_Spec *ASCIIToUnicodeCastSpec =
+            malloc(sizeof(PyArrayMethod_Spec));
+
+    ASCIIToUnicodeCastSpec->name = a2u_name;
+    ASCIIToUnicodeCastSpec->nin = 1;
+    ASCIIToUnicodeCastSpec->nout = 1;
+    ASCIIToUnicodeCastSpec->flags = NPY_METH_SUPPORTS_UNALIGNED;
```

Comment:
You don't need to support unaligned here, numpy will chain the cast if necessary, and it should be relatively rare (and if not, we should probably fix that part and not this here).